### DISCUSSION TITLE
Paginate relations

### DISCUSSION THREAD

**Discussion by codedge** [filamentphp/filament#380] [Help] [answered]
I was adding multiple relations which is perfectly described in the docs 👍 
One thing is currently unclear: Is pagination of relations not supported or is there are configuration to enabled this? It currently render every related entry which can cause some trouble 😅

**Comment by ryangjchandler** [comment]
> I was adding multiple relations which is perfectly described in the docs 👍
> One thing is currently unclear: Is pagination of relations not supported or is there are configuration to enabled this? It currently render every related entry which can cause some trouble 😅

Pagination should be automatic after 25 records, I believe.

**Reply by codedge** [reply]
I cannot confirm this, at least not for relations. For the repositories you are right. It automatically paginates with default 25.
But for some relation I have, there is a table with all the related entries rendered which are 99 rows - which is the correct amount of result but not paginated.

**Reply by zltn** [reply]
I had a similar problem couple weeks back,believe pagination on relationship managers is disabled by default (Livewire limitation that relies on querystrings).

If you only have a single relation you can enable it by publishing the relationship manager view and add below somewhere in `<x-tables::table />`

```blade
@if ($this->getTable()->hasPagination())
    <x-tables::pagination.paginator :paginator="$records" />
@endif
```
In the relationship manager you also need to:
1) overwrite `render()` to point to your own view 
2) add a `public $recordsPerPage = 10; //or what ever number of results you'd like to show`

Believe this will be configurable in 2.0 but @danharrin can confirm

**Comment by gavinhewitt** [comment, marked as answer]
> I was adding multiple relations which is perfectly described in the docs 👍
> One thing is currently unclear: Is pagination of relations not supported or is there are configuration to enabled this? It currently render every related entry which can cause some trouble 😅

Having the same issue :-(

**Reply by codedge** [reply]
So I guess this is a general issue then(?)

@zltn 
I have multiple relations (3) for my resource. So I would rather have this fixed in general. 

@gavinhewitt 
Were you able to find anything about this issue or know where it maybe comes from?

**Reply by danharrin** [reply]
Relations are not paginated by default. This is due to a limitation with Livewire that only allows you to paginate one collection on each page. Since you can have multiple relation managers attached to a resource, pagination is disabled to ensure this limitation is not hit.

To enable it, override the ListRecords class on your relation manager with your own, and enable pagination on that.

**Reply by codedge** [reply]
There seems to be a bit more to do:

- Overriding the the `$components['list']` entry with your custom `ListRecords` class which extends the base `\Filament\Resources\RelationManager\ListRecords`
- Overriding `hasPagination()` to return `true`

This paginates the records but does not show any pagination links under the relations table as inside the view `resources/views/relation-manager.blade.php` there is no placeholder for any pagination. While this helps at first, it does not let the user paginate the table(s) of the relations.

To override the `resources/views/relation-manager.blade.php` view with your own you can override the `render` method in my relation managers class. But what about the records that are needed for pagination.
There is a `CanGetRecords` trait which seems the provide the records via the `getRecords()` method, but there seems to be a lot more to be implemented to make this work with relations.

The issue seems that Livewire paginate the tables in the query string by using a `page` parameter. So you need to make sure that when having multiple paginated tables on one page - which is the case when having multiple relation managers on one resource - you have different parameters for the query.

**Reply by danharrin** [reply]
@codedge Feel free to PR in the paginator to `resources/views/resources/relation-manager/list-records.blade.php` 👍
